### PR TITLE
Iotbzh/for upstream/i2c

### PR DESCRIPTION
**Pull request by aaillet**

### CODE REVIEW

File: drivers/i2c/i2c_rcar.c [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+/* Bits to manage ICMIER and ICMSR registers */
+#define RCAR_I2C_MNR            BIT(6)  /* Master Nack Received */
+#define RCAR_I2C_MAL            BIT(5)  /* Master Arbitration lost */
+#define RCAR_I2C_MST            BIT(4)  /* Master Stop Transmitted */
+#define RCAR_I2C_MDE            BIT(3)  /* Master Data Empty */
+#define RCAR_I2C_MDT            BIT(2)  /* Master Data Transmitted */
+#define RCAR_I2C_MDR            BIT(1)  /* Master Data Received */
+#define RCAR_I2C_MAT            BIT(0)  /* Master Address Transmitted */
+
+/* Recommended settings from official documentation */
```
Comment (julien-massot) [inline drivers/i2c/i2c_rcar.c]:
```suggestion
/* Recommended bitrate settings from official documentation */
```

File: drivers/i2c/i2c_rcar.c [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+{
+	sys_write32(value, config->reg_addr + offs);
+}
+
+static void i2c_rcar_isr(const struct device *dev)
+{
+	const struct i2c_rcar_cfg *config = DEV_I2C_CFG(dev);
+	struct i2c_rcar_data *data = DEV_I2C_DATA(dev);
+
+	if (((i2c_rcar_read(config, RCAR_I2C_ICMSR)) & data->status_mask) == data->status_mask) {
```
Comment (julien-massot) [inline drivers/i2c/i2c_rcar.c]: this line is a bit long you can do
	if (((i2c_rcar_read(config, RCAR_I2C_ICMSR)) & data->status_mask) == data->status_mask) {
```suggestion
	if (((i2c_rcar_read(config, RCAR_I2C_ICMSR)) & data->status_mask) == 
	      data->status_mask) {
```

File: drivers/i2c/i2c_rcar.c (hunk 1 above)
Comment (aaillet) [reply drivers/i2c/i2c_rcar.c]: Fixed, thanks

File: drivers/i2c/i2c_rcar.c (hunk 2 above)
Comment (aaillet) [reply drivers/i2c/i2c_rcar.c]: Fixed, thanks

**Comment by aaillet** [comment]
reworded some commits

**Comment by aaillet** [comment]
Upstream PR ongoing

**Pull request closed without merging**